data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -110,16 +114,35 @@ private List<S3EventNotification.S3EventNotificationRecord> convertS3EventMessag
     }
 
     private void processS3ObjectAndDeleteSqsMessages(final Map<Message, List<S3EventNotification.S3EventNotificationRecord>> s3EventNotificationRecords) {
+        List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntryCollection = new ArrayList<>();
         for (Map.Entry<Message, List<S3EventNotification.S3EventNotificationRecord>> entry: s3EventNotificationRecords.entrySet()) {
             if (entry.getValue().isEmpty()) {
-                // TODO: delete or ignore based on configuration
+                if (s3SourceConfig.getOnErrorOption().equals(OnErrorOption.DELETE_MESSAGES)) {
+                    deleteMessageBatchRequestEntryCollection.add(buildDeleteMessageBatchRequestEntry(entry.getKey()));
+                }
```

Comment:
In line 105 and 110 we return an `Collections.emptyList()` when we can't convert `Message` to `s3EventNotification` or when there is an `SdkClientException`.  We handle the empty list later in this function in line 119  by checking if its empty and deleting if `on_error` is `DELETE_MESSAGES`.

Reply:
Since a failure on `parseJson()` will result in an empty list, I see how this will end up working. It would be good to see some of this cleaned up in a future iteration.

Reply:
Thanks for helping clarify @dlvenable